components/resource_metering/src/recorder/summary.rs

```diff
@@ -0,0 +1,121 @@
+///
+/// `SummaryRecorder` uses some special methods ([record_read_keys]/[record_write_keys])
+/// to collect external statistical information.
+///
+/// See [SubRecorder] for more relevant designs.
+///
+/// [SubRecorder]: crate::recorder::SubRecorder
+#[derive(Default)]
+pub struct SummaryRecorder;
+
+impl SubRecorder for SummaryRecorder {
+    fn collect(
+        &mut self,
+        records: &mut RawRecords,
+        thread_stores: &mut HashMap<usize, LocalStorage>,
+    ) {
+        thread_stores.iter_mut().for_each(|(_, ls)| {
+            let summary = { std::mem::take(&mut *ls.summary_records.lock().unwrap()) };
+            for (k, v) in summary {
+                records.records.entry(k).or_default().merge_summary(&v);
+            }
+            // The request currently being polled has not yet been merged into the hashmap,
+            // so it needs to be processed separately. (For example, a slow request that is
+            // blocking needs to reflect in real time how many keys have been read currently)
+            if let Some(t) = ls.attached_tag.load_full() {
+                if t.extra_attachment.is_empty() {
+                    return;
+                }
+
+                let s = ls.summary_cur_record.take_and_reset();
+                records.records.entry(t).or_default().merge_summary(&s);
+            }
+        });
+    }
+}
+
+/// An implementation of [SubRecorder] for dispatch config.
+///
+/// The purpose of designing this recorder is that the summary module has
+/// a `summary_enable` field in the thread-local of each thread to control
+/// the switch of summary collection. We need a way to pass the latest
+/// configuration to thread-local of all threads in real time. This requirement
+/// can be met by implementing [SubRecorder].
+///
+/// [SubRecorder]: crate::recorder::SubRecorder
+#[derive(Clone)]
+pub struct ConfigDispatchRecorder {
+    cur_cfg: Arc<Mutex<Config>>,
+    switch_changed: Arc<AtomicBool>,
+}
+
+impl SubRecorder for ConfigDispatchRecorder {
+    fn collect(
+        &mut self,
+        _records: &mut RawRecords,
+        thread_stores: &mut HashMap<usize, LocalStorage>,
+    ) {
+        if !self.switch_changed.load(SeqCst) {
+            return;
+        }
+        self.switch_changed.store(false, SeqCst);
+        let enabled = {
+            let g = self.cur_cfg.lock().unwrap();
+            g.enabled
+        };
+        thread_stores.iter().for_each(|(_, ls)| {
+            ls.summary_enable.store(enabled, SeqCst);
+        })
+    }
+}
+
+impl ConfigDispatchRecorder {
+    pub fn new(cfg: Config) -> Self {
+        Self {
+            cur_cfg: Arc::new(Mutex::new(cfg)),
+            switch_changed: Arc::new(AtomicBool::new(false)),
+        }
+    }
+
+    pub fn dispatch(&self, cfg: Config) {
+        let mut g = self.cur_cfg.lock().unwrap();
+        let switch_changed = g.enabled != cfg.enabled;
+        *g = cfg;
+        self.switch_changed.store(switch_changed, SeqCst);
+    }
+}
```

Comment:
Why not doing this in `Recorder.handle_pause`?

Reply:
It seems to be a better way, but we may need to modify the interface of `SubRecorder` to pass `thread_stores`. I'll make update.